docker/build-tools/Dockerfile

```diff
@@ -57,30 +62,33 @@ RUN go get github.com/jteeuwen/go-bindata/go-bindata
 RUN go get istio.io/tools/protoc-gen-docs
 RUN go get istio.io/tools/cmd/annotations_prep
 
-# Save some bytes in the container
-RUN upx --lzma /go/bin/*
+# Put the stuff we need in its final output location
+RUN mkdir -p ${OUTDIR}/usr/bin
+RUN mkdir -p ${OUTDIR}/usr/include
+RUN mkdir -p ${OUTDIR}/usr/local/go/bin
+RUN mkdir -p ${OUTDIR}/usr/local/go/pkg/mod
 
-# Cleanup
-WORKDIR /
-RUN apt-get purge -y \
-    autoconf \
-    libc-ares-dev \
-    libssl-dev \
-    libtool \
-    unzip \
-    upx \
-    zlibc \
-    zlib1g-dev \
-    && apt-get autoremove -y \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* \
-    && rm -fr /home /tmp/* \
-    && go clean -cache \
-    && go clean -modcache
+#RUN install -c ${GOPATH}/bin/* ${OUTDIR}/usr/bin
+RUN cp -aR /tmp/bin/protoc ${OUTDIR}/usr/local/go/bin
+RUN cp -aR /tmp/include/* ${OUTDIR}/usr/include
+RUN cp -aR /go/bin ${OUTDIR}/usr/local/go/bin
+RUN cp -aR /usr/local/go/bin/* ${OUTDIR}/usr/local/go/bin
+
+# Use inplace decompression on the go toolchain binaries
+#RUN upx --lzma ${OUTDIR}/usr/bin/*
+
+FROM golang:1.12 as build_tools_mod
 
 # Setup to run the Go compiler with host-side build cache, test cache, and module cache
+FROM ubuntu:xenial
+COPY --from=build_tools_context /out/ /
+COPY --from=build_tools_mod /usr/local/go/pkg/ /usr/local/go/pkg/
+
+RUN apt-get update
+RUN apt-get install -y make ca-certificates git
+
 ENV GOCACHE=/gocache
-RUN rm -fr /go/pkg/mod
+ENV PATH=/usr/local/go/bin:$PATH
 
```

Comment:
Given this isn't based on the golang image, I think we need to set GOPATH here, no?